### DISCUSSION TITLE
Write a langium grammar based on an exisiting source (meta) model?

### DISCUSSION THREAD

**Discussion by PatrikN-work** [eclipse-langium/langium#1426] [Q&A]
Hi!

is it possible to write aLangium grammar based on an exisiting source (meta) model? Similar to whta is possible in xtext refering to an existing ecore model.

I'de like to develop a grammar for already existing models

**Comment by msujew** [comment]
Hey @PatrikN-work,

You can [declare meta models](https://langium.org/docs/sematic-model/#declared-types) similar to EMF. We have our own syntax for it, that is mostly copied from TypeScript. 

> I'de like to develop a grammar for already existing models

Note that we don't support any other source of meta model definitions.
